### PR TITLE
Adding a new table to the OpenBR report

### PR DESCRIPTION
**Pull request by stephenrawls**
Adding a new table to the OpenBR report showing FAR values at the following fixed TAR values: 0.95, 0.85, 0.75, 0.65, 0.50, 0.40.

This is useful because reports for the IARPA project JANUS require us to report FAR at TAR=0.85.

Note that because I introduced a new function called getOperatingPointGivenFAR, I renamed the existing getOperatingPoint function to getOperatingPointGivenTAR to both be more explicit and to match the naming style.

### CODE REVIEW

File: openbr/core/plot.cpp [hunk 1]
```diff
@@ -249,6 +261,8 @@ struct RPlot
                                "plot.new()\n"
                                "print(textplot(FTtable))\n"
                                "print(title(\"Table of True Accept Rates at various False Accept Rates\"))\n"
+	                       "print(textplot(F_at_Ttable))\n"
```
Comment (jklontz) [inline openbr/core/plot.cpp]: indentation

File: openbr/core/eval.cpp [hunk 2]
```diff
@@ -288,13 +315,21 @@ float Evaluate(const Mat &simmat, const Mat &mask, const QString &csv, const QSt
...
+    lines.append(qPrintable(QString("FT,0.01,%1").arg(QString::number(getOperatingPointGivenFAR(operatingPoints, 0.01).TAR, 'f', 3))));
+    lines.append(qPrintable(QString("FT,0.1,%1").arg(QString::number(getOperatingPointGivenFAR(operatingPoints, 0.1).TAR, 'f', 3))));
+
+    // Write FAR@TAR Table (FatT)
+    lines.append(qPrintable(QString("FatT,0.95,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.95).FAR, 'f', 3))));
+    lines.append(qPrintable(QString("FatT,0.85,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.85).FAR, 'f', 3))));
+    lines.append(qPrintable(QString("FatT,0.75,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.75).FAR, 'f', 3))));
+    lines.append(qPrintable(QString("FatT,0.65,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.65).FAR, 'f', 3))));
+    lines.append(qPrintable(QString("FatT,0.50,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.50).FAR, 'f', 3))));
+    lines.append(qPrintable(QString("FatT,0.40,%1").arg(QString::number(getOperatingPointGivenTAR(operatingPoints, 0.40).FAR, 'f', 3))));
```
Comment (jklontz) [inline openbr/core/eval.cpp]: Do you mind refactoring the parts where we write tables to use foreach loops over the constant values? I understand you were just following the current convention, but the code duplication is getting out of hand :)

    foreach (float val, QList<float>() << 0.95 << 0.85 << 0.75 << 0.65 << 0.5 << 0.4)
      lines.append(/*...*/);

**Comment by jklontz** [comment]
LGTM otherwise. @bhklein please review

File: openbr/core/eval.cpp [hunk 3]
```diff
@@ -75,6 +75,32 @@ static OperatingPoint getOperatingPoint(const QList<OperatingPoint> &operatingPo
     return OperatingPoint(mScore * FAR + bScore,FAR, mTAR * FAR + bTAR);
 }
 
+static OperatingPoint getOperatingPointGivenTAR(const QList<OperatingPoint> &operatingPoints, float TAR)
+{
+    int index = 0;
+    while (operatingPoints[index].TAR < TAR) {
+      index++;
+      if (index == operatingPoints.size())
+	return OperatingPoint(operatingPoints.last().score, operatingPoints.last().FAR, TAR);
```
Comment (bhklein) [inline openbr/core/eval.cpp]: Indent instead of space

File: openbr/core/plot.cpp [hunk 4]
```diff
@@ -205,6 +207,16 @@ struct RPlot
...
                        "\n"
+                       "# Code to format TAR@FAR table\n"
+                       "algs <- unique(FT$%2)\n"
+                       "algs <- algs[!duplicated(algs)]\n"
+                       "mat <- matrix(FatT$Y,nrow=6,ncol=length(algs),byrow=FALSE)\n"
+                       "colnames(mat) <- algs \n"
+                       "rownames(mat) <- c(\"TAR = 0.95\", \"TAR = 0.85\", \"TAR = 0.75\", \"TAR = 0.65\", \"TAR = 0.50\", \"TAR = 0.40\")\n"
+                       "F_at_Ttable <- as.table(mat)\n"
+                       "\n"
+
```
Comment (bhklein) [inline openbr/core/plot.cpp]: remove blank line and one newline character here

**Comment by bhklein** [comment]
Looks good to me after the indentation fixes, thanks for the contribution @stephenrawls

**Comment by stephenrawls** [comment]
I just committed and pushed changes to my indentation and code style as advised by comments. I think you can see them now, but let me know if I need to do anything else to the pull request. (This is my first time using github...). Thanks.

File: openbr/core/eval.cpp [hunk 5]
```diff
@@ -288,13 +315,31 @@ float Evaluate(const Mat &simmat, const Mat &mask, const QString &csv, const QSt
...
-    lines.append(qPrintable(QString("FT,0.001,%1").arg(QString::number(getOperatingPoint(operatingPoints, 0.001).TAR, 'f', 3))));
-    lines.append(qPrintable(QString("FT,0.01,%1").arg(QString::number(getOperatingPoint(operatingPoints, 0.01).TAR, 'f', 3))));
-    lines.append(qPrintable(QString("FT,0.1,%1").arg(QString::number(getOperatingPoint(operatingPoints, 0.1).TAR, 'f', 3))));
+    // Write TAR@FAR Table (FT)
+    foreach (float far, QList<float>() << 1e-6 << 1e-5 << 1e-4 << 1e-3 << 1e-2 << 1e-1)
+      lines.append(qPrintable(QString("FT,%1,%2").arg(
+						      QString::number(far, 'f'),
+						      QString::number(getOperatingPointGivenFAR(operatingPoints, far).TAR, 'f', 3))));
+
+//    lines.append(qPrintable(QString("FT,0.000001,%1").arg(QString::number(getOperatingPointGivenFAR(operatingPoints, 0.000001).TAR, 'f', 3))));
```
Comment (jklontz) [inline openbr/core/eval.cpp]: Please remove these commented sections, then we're good to merge.

**Comment by stephenrawls** [comment]
Ah yes. I meant to do that before the first commit, but it slipped my mind. Done.

**Comment by jklontz** [comment]
thanks

**Pull request merged**